crates/polars-plan/src/plans/optimizer/projection_pushdown/mod.rs

```diff
@@ -436,7 +436,7 @@ impl ProjectionPushDown {
                             &acc_projections,
                             expr_arena,
                             &file_info.schema,
-                            scan_type.sort_projection(&file_options) || hive_parts.is_some(),
```

Comment:
revert sorting projections in the projection pushdown optimizer from https://github.com/pola-rs/polars/commit/29373d1b249a49887edd5ac6cb4ea64983ca9969

Reply:
Maybe we should just project unsorted columns in the optimizer. That makes IO code a lot simpler.

Reply:
I have changed this for Parquet. I've left the other scans as they may rely on sorted projections.